lib/dal/src/workspace_snapshot.rs

```diff
@@ -1278,4 +1241,119 @@ impl WorkspaceSnapshot {
 
         Ok(did_dispatch)
     }
+
+    async fn find_existing_dependent_value(
+        &self,
+        value_id: Ulid,
+    ) -> WorkspaceSnapshotResult<(Ulid, Option<Ulid>)> {
+        let dv_category_id = self
+            .get_category_node(None, CategoryNodeKind::DependentValueSet)
+            .await?;
+
+        for dv_node_idx in self
+            .outgoing_targets_for_edge_weight_kind(dv_category_id, EdgeWeightKindDiscriminants::Use)
+            .await?
+        {
+            let dv_value_node_weight = self
+                .get_node_weight(dv_node_idx)
+                .await?
+                .get_dependent_value_node_weight()?;
+
+            if value_id == dv_value_node_weight.value_id() {
+                return Ok((dv_category_id, Some(dv_value_node_weight.id())));
+            }
+        }
+
+        Ok((dv_category_id, None))
+    }
+
+    pub async fn add_dependent_value(
+        &self,
+        change_set: &ChangeSet,
+        value_id: impl Into<Ulid>,
+    ) -> WorkspaceSnapshotResult<()> {
+        let value_id = value_id.into();
+        let (dv_category_id, existing_value_id) =
+            self.find_existing_dependent_value(value_id).await?;
+
+        if existing_value_id.is_some() {
+            return Ok(());
```

Comment:
Didn't we want to "touch" the existing node somehow to force conflict resolution?  In the case where there's a DVU running that hasn't had the removal of these go through the rebaser, and we modify something (that won't be visible to the running DVU), we don't want to lose the fact that things still need to run. The conflict should always be a "modified a deleted" (or "deleted a modified") one, and the resolution would be to keep the modified version. In the case of a "conflicting modify", we can deduplicate down to one (doesn't really matter which). Same if we detect that two change sets tried to add the same one when neither started from a base that had it.